methods.py

```diff
@@ -1048,6 +1048,40 @@ def progress_finish(target, source, env):
     AlwaysBuild(progress_finish_command)
 
 
+def validate_target_filename(suffix):  # Check if target file can be written to.
+    if os.access("bin/godot" + suffix, os.F_OK) and not os.access("bin/godot" + suffix, os.W_OK):
```

Comment:
How about reversing the condition and adding eager `return` in case the original name is available, so the whole function does not need to be indented needlessly.

```
if not os.access("bin/godot" + suffix, os.F_OK) or os.access("bin/godot" + suffix, os.W_OK):
    return suffix
```

Reply:
Thanks!